stl/inc/deque

```diff
@@ -557,9 +555,21 @@ public:
 
     _Deque_val() noexcept : _Map(), _Mapsize(0), _Myoff(0), _Mysize(0) {}
 
-    size_type _Getblock(size_type _Off) const noexcept {
+    _Map_difference_type _Getblock(size_type _Off) const noexcept {
         // NB: _Mapsize and _Block_size are guaranteed to be powers of 2
-        return (_Off / _Block_size) & (_Mapsize - 1);
+        return static_cast<_Map_difference_type>((_Off / _Block_size) & (_Mapsize - 1));
+    }
+
+    reference _At(size_type _Idx) noexcept {
+        const auto _Block = _Getblock(_Idx);
+        const auto _Off   = static_cast<difference_type>(_Idx % _Block_size);

```

Comment:
If I'm correct, this `difference_type` come from `_Alty_traits::difference_type`, which has a different source from `_Map_difference_type`.
My suggestions are:
1. not changing return type of `_Getblock`.
2. this function should be
```
const size_type _Block = _Getblock(_Idx);
const auto _Off            = static_cast<_Map_difference_type >(_Idx % _Block_size);
return _Map[_Block][_Off];
```

Reply:
> If I'm correct, this `difference_type` comes from `_Alty_traits::difference_type`, which has a different source from `_Map_difference_type` (`_Al[p]ty_traits`).

This is right.

> My suggestions are:
> 
> 1. not changing return type of `_Getblock`.
> 2. this function should be
> 
> ```
> const size_type _Block = _Getblock(_Idx);
> const auto _Off        = static_cast<_Map_difference_type>(_Idx(or "_Off") % _Block_size);
> return _Map[_Block][_Off];
> ```

These suggestions don't seem correct. The new test case will be rejected with these applied.

Reply:
> The new test case will be rejected with these applied.

Yes, I mistook the types here :(